app/services/course/koditsu_workspace_service.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+class Course::KoditsuWorkspaceService
+  def initialize(course)
+    @course = course
+
+    @course_object = { name: "#{@course.id}_#{course.title}" }
+  end
+
+  def run_create_koditsu_workspace_service
+    return if @course.koditsu_workspace_id
+
+    koditsu_api_service = KoditsuAsyncApiService.new('api/workspace', @course_object)
+    response_status, response_body = koditsu_api_service.post
+
+    unless response_status == 201
+      raise KoditsuError,
+            { status: response_status, body: response_body }
+    end
+
+    response_body['data']
```

Comment:
Personally I feel that you did not standardize your function signatures.  
Some return `[status, data]  `
while this one just returns `data`  

The discrepancy introduces additional overheads for developer to remember.

Reply:
status is only returned if it actually makes sense in the whole greater scheme of why this function is called in the first place.. the reason I did not return status here is because if it fails, the error is raised even before reaching this point

on the other hand, in the other places (let's say, creating question), there are 2 APIs involved here; create question and edit assessment in koditsu.. this case, status needs to be returned since it's needed to decide what to do with the subsequent API being called

though, yes you're right that it might introduce additional overheads.. in this case, I'll do refactoring after this PR (quite too many needs to be adjusted if we standardize it now)